lib/runner/request-helpers-presend.js

```diff
@@ -12,20 +12,38 @@ module.exports = [
         payload._item = new sdk.Item(payload._item.toObjectResolved(null,
             [payload.data, payload.environment.syncVariablesTo({}), payload.globals.syncVariablesTo({})]));
         // todo: call before variable resolution triggers on the run object
+
+        var item = payload._item;
+
+        // Re-parse the URL, because variables have been resolved now, and things might be moved around
+        item.request.url = new (sdk.Url)(item.request.url.toString());
+
         done();
     },
 
     // Authorization
     function (payload, run, done) {
         if (!payload._item) { return done(new Error('Nothing to authorize.')); }
 
-        var item = payload._item;
-        // Re-parse the URL, because variables have been resolved now, and things might be moved around
-        item.request.url = new (sdk.Url)(item.request.url.toString());
+        var item = payload._item,
+            cursor = payload.coords;
 
-        // perform the actual signature additions etc
-        item.request = item.request.authorize();
-        done();
+        async.waterfall([
+            // Pre verify
+            function (cb) {
+                run.authorizer.preVerify(item, run, cursor, cb);
+            },
+            // Call initialize if necessary
+            function (verified, cb) {
+                verified ? cb(null) : run.authorizer.init(item, run, cursor, function (err) {
```

Comment:
Let's check if `cb` can be passed directly as a callback to `run.authorizer.init` here.